cli/pcluster/config/s3_param_types.py

```diff
@@ -0,0 +1,241 @@
+        # Convert value to Int
+        return int(value)
+
+
+class BooleanS3Param(S3Param):
+    """Base S3Param to manage boolean parameters."""
+
+    def _parse_value(self, value):
+        """Parse the value from config file, converting to the needed type for the specific param."""
+        # Convert value to boolean
+        return value and value.lower().strip() == "true"
+
+
+class FloatS3Param(S3Param):
+    """Base S3Param to manage float parameters."""
+
+    def _parse_value(self, value):
+        """Parse the value from config file, converting to the needed type for the specific param."""
+        # Convert value to Float
+        return float(value) if value else None
+
+
+class ComputeInstanceTypeS3Param(S3Param):
+    """S3Param to manage compute instance type in compute_resource sections."""
+
+    def refresh(self):
+        """
+        Populate additional settings needed for the linked compute resource like vcpus, gpus etc.
+
+        These parameters are set according to queue settings and instance type capabilities.
+        """
+        instance_type = utils.get_instance_type(self.value)
+
+        # Set vcpus according to queue's disable_hyperthreading and instance features
+        queue_section = self.owner_section.parent_section
+        ht_disabled = queue_section.get_param_value("disable_hyperthreading") if queue_section else None
+        vcpus_info = instance_type.get("VCpuInfo")
+        default_cores = vcpus_info.get("DefaultCores")
+        vcpus = default_cores if ht_disabled and default_cores else vcpus_info.get("DefaultVCpus")
+        self.owner_section.get_param("vcpus").value = vcpus
+
+        # Set gpus according to instance features
+        gpu_info = instance_type.get("GpuInfo", None)
+        if gpu_info:
+            # TODO: check gpu retrieving logic
+            self.owner_section.get_param("gpus").value = sum([gpus.get("Count") for gpus in gpu_info.get("Gpus")])
+
+        # Set enable_efa according to queues' enable_efa and instance features
+        enable_efa = queue_section.get_param_value("enable_efa") if queue_section else None
+        self.owner_section.get_param("enable_efa").value = enable_efa and instance_type.get("NetworkInfo").get(
+            "EfaSupported"
+        )
+
+
+class ScaleDownIdleTimeS3Param(S3Param):
+    """S3Param to manage scaledown_idletime for S3 configuration."""
+
+    def refresh(self):
+        """Take the value from the scaledown_idletime cfn parameter."""
+        self.value = self.owner_section.get_param("scaledown_idletime").value
+
+    def get_storage_key(self):
+        """Return the key by which the current param must be stored in the JSON."""
+        return "scaledown_idletime"
+
+
+class DefaultComputeQueueS3Param(S3Param):
+    """S3Param to manage default_queue parameter in cluster section."""
+
+    def refresh(self):
+        """Take the label of the first queue as value."""
```

Comment:
We're relying on the order of the queues to set the "default_queue".
How can the user change the default queue?

What does the "default_queue" imply?

Reply:
This is per design. In the initial implementation the default queue will be the first one, and changing the order of the queues will change the default queue. This behaviour could be re-discussed in following releases.